lib/galaxy/tools/deps/resolvers/conda.py

```diff
@@ -267,11 +271,16 @@ def version(self):
         return self._version
 
     def shell_commands(self, requirement):
-        return """[ "$CONDA_DEFAULT_ENV" = "%s" ] || . %s '%s' 2>&1 """ % (
-            self.environment_path,
-            self.activate,
-            self.environment_path
-        )
+        if self._preserve_python_environment:
+            return """export PATH=$PATH:'%s/bin' """ % (
```

Comment:
I know that it is broken in some cases (e.g. Python modules) - but it works for samtools. I think PATH is good enough - but I'm open to doing more if you have specific ideas.